src/CodeGen_Hexagon.cpp

```diff
@@ -1286,6 +1286,7 @@ string CodeGen_Hexagon::mattrs() const {
     if (target.has_feature(Halide::Target::HVX_128)) {
         attrs << ",+hvx-double";
     }
+    attrs.seekp(0, std::ios_base::end);
```

Comment:
I think the fix should be to rewrite this method:

    std::stringstream attrs;
    attrs << "+long-calls";
    if (target.has_feature(Target::HVX_64)) {
        attrs << ",+hvx";
    } else if (target.has_feature(Target::HVX_128)) {
        attrs << ",+hvx-double";
    }

Note it changes the behavior (we used to always use +hvx, now we wouldn't).